### ISSUE TITLE
[FR]: Update feeds via command

### ISSUE THREAD

**Issue by Spixmaster** [martinrotter/rssguard#1327]
### Brief description of the feature request

I would like to update my feeds with a command as I want to automate the updating with a script on a regular basis.

The reasoning behind this is that some feeds only show the last ten entries like WordPress and I do not want to miss feeds which unfortunately already happened to me. It is not possible to get these articles later once they are gone.

**Comment by martinrotter** [comment]
What you mean "with command" like console command? You must be much more specific, my friend.

**Comment by Spixmaster** [comment]
@martinrotter Yes, I mean a console command like `rssguard fetch`, an option would be fine too. It is your choice. This would be the most efficient way to do so. Otherwise the whole GUI needs to be started and automation is not possible.

**Comment by martinrotter** [comment]
Well, RSS Guard is GUI application and CLI automation is not really there.

BUT, that said. Once RSS Guard is configured, then with a bit of work it could be done that it starts and runs in headless mode. I will investigate the possibility.

**Comment by martinrotter** [comment]
No, I plan/want to make headless mode fully working. No windows, no dialogs, no popups.

I had this in my head for some time, would work like this:

1. You launch your RSS Guard normally with GUI and setup all your feeds and their updating intervals etc., then you quit RSS Guard.
2. You launch now RSS Guard with "headless" process parameter switch and RSS Guard will run in non-GUI mode (it will not display main window, it will not show any popups etc.)

THis is not problem about parsing CLI arguments etc. that functionality is already in RSS Guard and working

**Comment by Spixmaster** [comment]
Would the pure CLI have the functionality to just fetch the new feeds?

Is it this function, https://github.com/martinrotter/rssguard/blob/310c3068bda2702f6d485835f2d28df4cb4550e4/src/librssguard/core/feeddownloader.cpp#L69C10-L69C33? Would it be sufficient to call it with a command?

**Comment by martinrotter** [comment]
That depends, Either it will be the same executable file but will behave "headless" when proper switch is invoked and it will behave exactly the same as GUI version (so download feeds regularly per settings, etc.) but will not show any GUI at all.

OR

I can write brand new "CLI ONLY" executable file for this purpose only.

**Comment by martinrotter** [comment]
I would likely be "cleaner" to have separate "rssguard_cli.exe" for CLI-only application. It will use "librssguard.dll" library which contains all needed stuff and will call only needed functions.

**Comment by Spixmaster** [comment]
I do not know whether that would contain the feature to updated the feeds, does it?

**Comment by Spixmaster** [comment]
Do not you think that something like `rssguard fetch` or `rssguard-cli fetch` which updated all feeds would be useful?

**Comment by martinrotter** [comment]
> I do not know whether that would contain the feature to updated the feeds, does it?

Of course it will contain the feature. If you have discord, please join rssguard discord channel and we can chat this.

https://discord.com/invite/7xbVMPPNqH

**Comment by Spixmaster** [comment]
That is great. Thanks.

**Comment by martinrotter** [comment]
I myself do not plan fully implementing this, PRs are welcomed tho. :)

To fetch feeds regularly, just set proper auto-fetch interval for them.
